ecosystem/sf-indexer/aptos-substreams/src/token_converter.rs

```diff
@@ -0,0 +1,369 @@
+                "version {} failed! collection creator resource was missing, table handle {} not in map {:?}",
+                txn_version, standardize_handle(&table_item.handle), table_handle_to_owner,
+            ))?
+            .clone();
+        return Ok(Some(CollectionData {
+            collection_name: value["name"]
+                .as_str()
+                .map(|s| s.to_string())
+                .context(format!(
+                    "version {} failed! name missing from collection {:?}",
+                    txn_version, value
+                ))?,
+            creator_address,
+            description: value["description"]
+                .as_str()
+                .map(|s| s.to_string())
+                .context(format!(
+                    "version {} failed! description missing from collection {:?}",
+                    txn_version, value
+                ))?,
+            transaction_version: txn_version,
+            metadata_uri: value["uri"]
+                .as_str()
+                .map(|s| s.to_string())
+                .context(format!(
+                    "version {} failed! uri missing from collection {:?}",
+                    txn_version, value
+                ))?,
+            supply: value["supply"]
+                .as_str()
+                .map(|s| s.parse::<u64>())
+                .context(format!(
+                    "version {} failed! supply missing from collection {:?}",
+                    txn_version, value
+                ))?
+                .context(format!(
+                    "version {} failed! failed to parse supply {:?}",
+                    txn_version, value["supply"]
+                ))?,
+            maximum: value["maximum"]
+                .as_str()
+                .map(|s| s.parse::<u64>())
+                .context(format!(
+                    "version {} failed! maximum missing from collection {:?}",
+                    txn_version, value
+                ))?
+                .context(format!(
+                    "version {} failed! failed to parse maximum {:?}",
+                    txn_version, value["maximum"]
+                ))?,
+            maximum_mutable: value["mutability_config"]["maximum"]
+                .as_bool()
+                .context(format!(
+                    "version {} failed! mutability_config.maximum missing {:?}",
+                    txn_version, value
+                ))?,
+            uri_mutable: value["mutability_config"]["uri"]
+                .as_bool()
+                .context(format!(
+                    "version {} failed! mutability_config.uri missing {:?}",
+                    txn_version, value
+                ))?,
+            description_mutable: value["mutability_config"]["description"]
+                .as_bool()
+                .context(format!(
+                    "version {} failed! mutability_config.description missing {:?}",
+                    txn_version, value
+                ))?,
+        }));
```

Comment:
Btw we should probably use constants for things that are common here, this seems pretty fragile

Reply:
Hmm yes but changing any of these would be a breaking change IMO.